cmd/signal/grpc/server/server.go

```diff
@@ -42,6 +43,7 @@ func NewServer(sfu *sfu.SFU) *SFUServer {
 // If the client closes this stream, the webrtc stream will be closed.
 func (s *SFUServer) Signal(stream pb.SFU_SignalServer) error {
 	peer := sfu.NewPeer(s.SFU)
+	var streamLock sync.Mutex
```

Comment:
why is this necessary?

Reply:
This is the error when i comment this lock:
`[2021-01-22 17:57:42.545] [ERROR] [132][peer.go][func4] => oniceconnectionstatechange error rpc error: code = Internal des
c = transport: transport: the stream is done or WriteHeader was already called`
The reason is:
https://github.com/grpc/grpc-go/blob/master/Documentation/concurrency.md#streams

Reply:
got it. i have seen that error. can we embed the lock in the sever instead?

```
type SFUServer struct {
	pb.UnimplementedSFUServer
        sync.Mutex
	SFU *sfu.SFU
}
```

Reply:
Good idea. 
there is only one recv goroutine, so we only need a send lock.